bee-common/src/logger/config.rs

```diff
@@ -0,0 +1,139 @@
+// Copyright 2020 IOTA Stiftung
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+// the License. You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+// an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and limitations under the License.
+
+use crate::logger::LOGGER_STDOUT_NAME;
+
+use log::LevelFilter;
+use serde::Deserialize;
+
+/// Default value for the color flag.
+const DEFAULT_COLOR: bool = true;
+/// Default name for an output.
+const DEFAULT_NAME: &str = LOGGER_STDOUT_NAME;
+/// Default log level for an output.
+const DEFAULT_LEVEL: &str = "info";
+
+/// Builder for a logger output configuration.
+#[derive(Default, Deserialize)]
+pub struct LoggerOutputConfigBuilder {
+    // Name of an output.
+    name: Option<String>,
+    // Log level of an output.
+    level: Option<String>,
+}
+
+impl LoggerOutputConfigBuilder {
+    /// Creates a new builder for a logger output configuration.
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    /// Sets the name of a logger output.
+    pub fn name(mut self, name: &str) -> Self {
+        self.name.replace(name.to_string());
+        self
+    }
+
+    /// Sets the level of a logger output.
+    pub fn level(mut self, level: &str) -> Self {
+        self.level.replace(level.to_string());
+        self
+    }
+
+    /// Builds a logger output configuration.
+    pub fn finish(self) -> LoggerOutputConfig {
+        let level = match self.level.unwrap_or_else(|| DEFAULT_LEVEL.to_owned()).as_str() {
+            "trace" => LevelFilter::Trace,
+            "debug" => LevelFilter::Debug,
+            "info" => LevelFilter::Info,
+            "warn" => LevelFilter::Warn,
+            "error" => LevelFilter::Error,
+            _ => LevelFilter::Info,
```

Comment:
as mentioned above....matching **any** other string to `Info` is not necessary. We shouldn't allow the user to mess with our internal state in the first place if we don't have to.

Reply:
Answered in the other comment.